plugins/kubernetes-backend/src/routes/resourcesRoutes.ts

```diff
@@ -65,25 +76,89 @@ export const addResourceRoutesToRouter = (
   };
 
   router.post('/resources/workloads/query', async (req, res) => {
+    const token = getBearerTokenFromAuthorizationHeader(
+      req.header('authorization'),
+    );
+
+    const authorizeResponse = (
+      await permissionApi.authorizeConditional(
+        [
+          {
+            permission: kubernetesWorkloadResourcesReadPermission,
+            resourceRef: req.body.id,
+          },
+        ],
+        { token },
+      )
+    )[0];
+
+    if (authorizeResponse.result === AuthorizeResult.DENY) {
+      res.status(403).json({ error: new NotAllowedError() });
+      return;
+    }
+
     const entity = await getEntityByReq(req);
     const response = await objectsProvider.getKubernetesObjectsByEntity({
       entity,
       auth: req.body.auth,
     });
-    res.json(response);
+    let editedResponse = response;
+
+    if (authorizeResponse.result === AuthorizeResult.CONDITIONAL) {
+      const permissionCondition =
+        authorizeResponse.conditions as PermissionCondition;
```

Comment:
To reframe the same question I asked above, what happens when the `conditions` returned by the permission API is one of the other members of the union: `AllOf`, `AnyOf` or `Not`?

Reply:
I plan to add a unit test that may capture a policy decision that has a generic member conditions type. Because I am currently casting the conditions returned in the policy decision I believe this response type would break this current implementation so I do plan to introduce some sort of condition transformer that can change the decision returned by our permissionapi call to something useable for filtering regardless of what conditions type is returned.